### PR TITLE
msgpack: add support for MessagePack v6.0.0

### PR DESCRIPTION
**Pull request by komainu8**
GitHub: fix GH-1536.
msgpack.pc renamed to msgpack-c.pc in MessagePack v6.0.0.

Reported Carlo Cabrera. Thanks!!!

### CODE REVIEW

**Review by kou — COMMENTED** [review]
Could you try `msgpack-c` first?

**Comment by komainu8** [comment]
Groonga can found MessagePack v6.0.0 as below.

CMake

See:  https://github.com/groonga/groonga/actions/runs/4422013191/jobs/7753402069#step:6:280

```
-- Checking for module 'msgpack-c'
--   Found msgpack-c, version 6.0.0
-- MessagePack: /usr/local/lib/libmsgpack-c.dylib
```

Autotools

See: https://github.com/groonga/groonga/actions/runs/4422013191/jobs/7753402010#step:5:687

```
   MessagePack:           yes
    CFLAGS:              -I/usr/local/Cellar/msgpack/6.0.0/include
    LIBS:                -L/usr/local/Cellar/msgpack/6.0.0/lib -lmsgpack-c
```

File: configure.ac [hunk 1]
```diff
@@ -1298,9 +1298,13 @@ AC_ARG_ENABLE(message-pack,
...
 if test "x$enable_message_pack" != "xno"; then
   m4_ifdef([PKG_CHECK_MODULES], [
-    PKG_CHECK_MODULES([MESSAGE_PACK], [msgpack],
+    PKG_CHECK_MODULES([MESSAGE_PACK], [msgpack-c],
       [message_pack_available=yes],
-      [message_pack_available=no])
+      [
+        PKG_CHECK_MODULES([MESSAGE_PACK], [msgpack],
+        [message_pack_available=yes],
+        [message_pack_available=no])
```
Comment (kou) [inline configure.ac:1306]:
```suggestion
          [message_pack_available=yes],
          [message_pack_available=no])
```

**Comment by carlocab** [comment]
Thank you, @komainu8, @kou!

**Comment by kou** [comment]
Thanks for upstreaming this!

**Pull request merged**